airflow/providers/google/cloud/operators/bigquery.py

```diff
@@ -1177,14 +1178,56 @@ def execute(self, context: Context):
             ]
         else:
             raise AirflowException(f"argument 'sql' of type {type(str)} is neither a string nor an iterable")
-        context["task_instance"].xcom_push(key="job_id", value=job_id)
+        context["task_instance"].xcom_push(key="job_id", value=self.job_id)
 
     def on_kill(self) -> None:
         super().on_kill()
         if self.hook is not None:
             self.log.info("Cancelling running query")
             self.hook.cancel_job(self.hook.running_job_id)
 
+    def get_openlineage_facets_on_complete(self, task_instance):
+        try:
+            from openlineage.client.facet import SqlJobFacet
+            from openlineage.common.provider.bigquery import BigQueryDatasetsProvider
+
+            from airflow.providers.openlineage.extractors import OperatorLineage
+            from airflow.providers.openlineage.utils.utils import normalize_sql
+        except ImportError:
+            return None
```

Comment:
This is a design choice here, both are possible. 

Thinking on it more, raising might be better, since this methods aren't supposed to be called outside of OpenLineage integration - and if OpenLineage integration calls it, the imports should work. So the raise situation is rather impossible outside of a bug.